packages/filesystem/src/browser/file-dialog/file-dialog.ts

```diff
@@ -69,6 +75,11 @@ export class FileDialogProps extends DialogProps {
      */
     filters?: FileDialogTreeFilters;
 
+    /**
+     * Allow to decide where the 'All Files' selector appears relative to the other filters
+     */
+    allFilesPosition?: FileDialogTreeAllFilesPosition;
```

Comment:
What is the use case? From an issue it seems that one wants to specify the default filter not a position of all files. @elaihau ?

How such APIs defined in Electron and VS Code API?

Reply:
I created that github issue for a use case where the user wanted to open a vscode / theia workspace file. 
In current Theia, the dialog always opens with the "All Files" filter. it would be nice if the `*.vscode-workspace` and `*.theia-workspace` filter can be used as the default filter when the dialog opens.

Reply:
`defaultFilter?: string` would be better, if not set or invalid, should fall back to `All Files`.